templates/model/inputs.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"gopkg.in/yaml.v3"
+)
+
+// This file parses a YAML file that describes template input.
+
+// InputValue represents one of the parsed "input" fields from the inputs.yaml file.
+type InputValue struct {
+	// Pos is the YAML file location where this object started.
+	Pos ConfigPos `yaml:"-"`
+
+	Name  String `yaml:"name"`
+	Value String `yaml:"value"`
+}
+
+type Inputs struct {
+	// Pos is the YAML file location where this object started.
+	Pos ConfigPos `yaml:"-"`
+
+	Inputs []*InputValue `yaml:"inputs"`
+}
+
+// DecodeInputs unmarshals the YAML Spec from r.
+func DecodeInputs(r io.Reader) (*Inputs, error) {
```

Comment:
As far as I can see, this function is only used in tests. Should it go in a _test.go file?

Reply:
We'll use that in test_record, test_render action in following up PRs. :)

Reply:
Oh, of course.